packages/flutter_tools/lib/src/application_package.dart

```diff
@@ -385,6 +385,11 @@ class BuildableIOSApp extends IOSApp {
   String get archiveBundlePath
     => globals.fs.path.join(getIosBuildDirectory(), 'archive', globals.fs.path.withoutExtension(_hostAppBundleName));
 
+  // The output xcarchive bundle path `build/ios/archive/Runner.xcarchive`.
+  String get archiveBundleOutputPath => globals.fs.path.setExtension(archiveBundlePath, '.xcarchive');
+
+  String get ipaOutputPath => globals.fs.path.join(getIosBuildDirectory(), 'ipa');
```

Comment:
It looks like the dimensions we have are device type and build modes. I suppose it's non-sensical to have debug ipas or simulator ipas. Was that the thinking process? Should user's own scheme/flavor be in there somewhere?

Reply:
That's what the `flutter build xcarchive` is for.  That was kind of my point about this IPA command being [kind of useless](https://github.com/flutter/flutter/issues/13065#issuecomment-698667902).  A lot of the requests in https://github.com/flutter/flutter/issues/13065 are solved by `flutter build xcarchive`, which has all the compiled code that encapsulates flavor, scheme, etc.
The build mode here only so the tool can find the right directory for their non-release archive and to find the right archive name.